coderd/aitasks.go

```
@@ -113,15 +113,27 @@ func (api *API) tasksCreate(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	taskName := taskname.GenerateFallback()
-	if anthropicAPIKey := taskname.GetAnthropicAPIKeyFromEnv(); anthropicAPIKey != "" {
-		anthropicModel := taskname.GetAnthropicModelFromEnv()
+	taskName := req.Name
+	if err := codersdk.NameValid(taskName); taskName != "" && err != nil {
```

I think it would be clearer to break out the `taskName != ""` check, it's a bit confusing that we're validating  "" and then ignoring the error.

---

Is the change in https://github.com/coder/coder/pull/19745/commits/3104048e78deaba67e006b2d3274709b03236047 better?